docs/source/python/extending_types.rst

```diff
@@ -286,6 +286,36 @@ are available).
 The same ``__arrow_ext_class__`` specialization can be used with custom types defined
 by subclassing :class:`ExtensionType`.
 
+Custom scalar conversion
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want scalars of your custom extension type to convert to a custom type when
+:meth:`ExtensionScalar.as_py()` is called, you can override the :meth:`ExtensionType.scalar_as_py()`
+method. For example, if we wanted the above example 3D point type to return a tuple instead
+of a list, we would implement::
```

Comment:
```suggestion
method. For example, if we wanted the above example 3D point type to return a custom
3D point class instead of a list, we would implement::
```